public/app/features/alerting/unified/utils/rules.ts

```diff
@@ -277,7 +277,13 @@ export const flattenCombinedRules = (rules: CombinedRuleNamespace[]) => {
     groups.forEach(({ name: groupName, rules }) => {
       rules.forEach((rule) => {
         if (rule.promRule && isAlertingRule(rule.promRule)) {
-          acc.push({ dataSourceName: getRulesSourceName(rulesSource), namespaceName, groupName, ...rule });
+          acc.push({
+            ...rule,
```

Comment:
somewhat concerned about the spread being moved to the start here as that _might_ impact how overlapping props are overwritten – any particular reason we've changed the spread order here?